cmd/agent/daemon/state/controller.go

```diff
@@ -87,13 +91,14 @@ func NewController(
 	ct conntrackClient,
 	tracer ebpfTracer,
 	signatureEngine signatureEngine,
-	enrichmentService enrichmentService,
 	kubeClient kubepb.KubeAPIClient,
 	processTreeCollector processTreeCollector,
 	procHandler procHandler,
+	eventsEnrichmentService eventsEnrichmentService,
 ) *Controller {
-	dnsCache, err := freelru.NewSynced[uint64, *freelru.SyncedLRU[netip.Addr, string]](1024, func(k uint64) uint32 {
-		return uint32(k) // nolint:gosec
+	dnsCache, err := freelru.NewSynced[netip.Addr, string](1024, func(addr netip.Addr) uint32 {
```

Comment:
We used to create lrus for each cgroup here for dns. I don't remember what is the issue to have single lru here 🤔 Maybe it's because of pod ips which may change. In such way we can can a single LRU but hash key digest by cgroup_id + addr.